hack/lib/util/misc.sh

```diff
@@ -233,3 +233,35 @@ function os::util::list_go_src_files() {
 	\) -name '*.go' | sort -u
 }
 readonly -f os::util::list_go_src_files
+
+# os::util::docker calls docker, elevating privileges if necessary.
+# Scripts that expect to need to elevate privilges should call
+# os::util::environment::use_sudo in their preamble.
```

Comment:
Previously the supposition was that you would not need root access for the Docker socket ... since we're not running a system that needs that level of security. We can update the `hack/lib/init.sh` preamble to check if the socket is write-able, I guess, and use that here.